virtcontainers/sandbox.go

```diff
@@ -1049,15 +1054,39 @@ func (s *Sandbox) addContainer(c *Container) error {
 	s.containers[c.id] = c
 
 	ann := c.GetAnnotations()
-	if ann[annotations.ContainerTypeKey] == string(PodSandbox) {
+	if ann[annotations.ContainerTypeKey] != string(PodSandbox) {
+		return nil
+	}
+
+	if !s.config.SandboxCgroupOnly {
+		// Use the same cgroup of the container sandbox as the sandbox cgroup
 		s.state.CgroupPath = c.state.CgroupPath
-		if !s.supportNewStore() {
-			return s.store.Store(store.State, s.state)
-		}
+		return nil
+	}
+
+	// Use the parent cgroup of the container sandbox as the sandbox cgroup
+	s.state.CgroupPath = filepath.Dir(c.state.CgroupPath) + "/kata-sandbox/"
```

Comment:
We are creating a cgroup, we should should remove it on delete (even if the caller will delete all the parent cgroup)
If we use a common name al the pods will use the same cgroup if the cgroup parent was not set just for the pod, e.g. docker. So probably need to add an additional sufix (container.id) or just drop kata in the parent cgroup instead  of create one.